js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1626,10 +1626,23 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
         }
     },
 
+    _allItemsExpanded: function() {
+        this._skipContentReadyAndItemExpanded = false;
+        this._fireContentReadyAction();
+    },
+
     expandAll: function() {
-        const dataAdapter = this._dataAdapter;
-        each(dataAdapter.getData(), (_, node) => dataAdapter.toggleExpansion(node.internalFields.key, true));
-        this.repaint();
+        const nodes = this._dataAdapter.getData();
+        const expandingPromises = [];
+
+        this._skipContentReadyAndItemExpanded = true;
+
+        // NOTE: This is needed to support animation on expandAll, but stop triggering lot od contentReady/itemExpanded events.
+        // Deleting _allItemsExpanded() used to avoid error when component is disposed before promises fullfilles.
```

Comment:
I think it's a default practice and it shouldn't be described with a note
```suggestion
```

Reply:
ok, deleted